app/js/profile_controller.js

```diff
@@ -9,26 +9,58 @@
   var OPERATORS = ['Movistar', 'MTN'];
 
   /**
+   * Get the list of selected handset types from the form element and returns
+   * them as an array.
+   * @param {array} selected_options - The selected handset types
+   * @returns {array} Selected handset types as an array
+   */
+  function get_selected_handset_types(selected_options) {
+    var handset_types = [];
+    for (var i = 0, l = selected_options.length; i < l; i++) {
+      handset_types.push(selected_options[i].value);
+    }
+    return handset_types;
+  }
+
+  /**
    * Collect information from the profile form on submission and
    * submits new data to server.
    */
   function update_user(evt) {
     evt.preventDefault();
 
-    var user_data = {};
+    var user = {};
+    var settings = [];
     var form = document.getElementById('profile');
-    var profile_elements = ['display_name', 'new_comments', 'new_questions',
-      'locale', 'handset_type', 'operator'];
+    var profile_elements = ['display_name', 'new_comment_notify',
+      'buddyup_reminder', 'locale', 'handset_type', 'operator'];
 
     for (var i = 0, l = profile_elements.length; i < l; i++) {
       var elem_name = profile_elements[i];
       var elem = form.elements[profile_elements[i]];
 
-      user_data[elem_name] = elem.type === 'checkbox' ?
-        elem.checked : elem.value;
+      switch(elem_name) {
+        case 'handset_type':
+          user[elem_name] = get_selected_handset_types(elem.selectedOptions);
+          break;
+        case 'new_comment_notify':
+        case 'buddyup_reminder':
+          settings.push({
+            'name': elem_name,
+            'value': elem.checked
+          });
+          break;
+        default:
+          user[elem_name] = elem.value;
```

Comment:
Not sure what you are commenting on here. We are sending all of the form fields to the server.